wsit/ws-sx/wssx-impl/src/main/java/com/sun/xml/wss/impl/misc/SecurityUtil.java

```diff
@@ -628,6 +632,49 @@ public static String getDataEncryptionAlgo(JAXBFilterProcessingContext context){
     public static URL loadFromContext(final String configFileName, final Object context) {
         return ReflectionUtil.invoke(context, "getResource", URL.class, configFileName);
     }
+
+
+    /**
+     * @param endpoint
+     * @return null or the ServletContext instance bound to this endpoint
+     */
+    public static Object getServletContext(final WSEndpoint<?> endpoint) {
```

Comment:
Wouldn't it throw an exception if the ServletContext class would not be on the classpath? It expects an Object on all places, then sometimes even calls method via reflection.

Reply:
see [type inference](https://docs.oracle.com/javase/tutorial/java/generics/genTypeInference.html) - in short as long as the type of the field the return value is assigned to remains Object, nothing changes; ideally, the servlet related code should be moved to some isolated place, since it is not required to be present (not something to be dealt with now).

Reply:
This would not help anything because everywhere is expected java.lang.Object and the result object is always used as it is without retyping.
I agree it should be moved somewhere, but it must be a consistent work in own PR.

Reply:
so it's better to leave it for the future instead of starting doing small steps towards the goal?

Reply:
This is not a step towards goal. Goal would be some separate class/module managing ServletContext if it is available. Less reflection, no need for this, it would simply work directly with the ServletContext class. Until that it is better to keep consistency.

Reply:
hm, are you saying that this consistency is good and consistency in avoiding static imports is bad?

Reply:
> this consistency is good 

because there is no advantage of using generics here

> consistency in avoiding static imports is bad

because there is no such consistency as there is another 100 static imports in the repository

Reply:
Are you going to fix those inconsistencies? If not then I would really appreciate you accept the fact that no new ones should be introduced. Thank you.

Reply:
> consistency in avoiding static imports is bad?

If they have never been avoided before, there is no new inconsistency right?

Also, why is not using static imports consistent? The project now also doesn't avoid imports. Isn't that not inconsistent too?

Reply:
@dmatej 
> I removed my static imports, and moved error messages.

Okay, it's a bit... questionable this entire request, but okay. @lukasj can this conversation now be resolved?

Reply:
> Also, why is not using static imports consistent?

as I mentioned earlier - during a reviews back in 2006-7 it was noted to be removed and avoided. There was a promise they'll be removed but as usually - people come and go and promise does not hurt, right?

I really do not understand what it is so difficult to just respect the request to avoid using it (2 lines change done in 2minutes?) and spent endless hours arguing why it is "the best feature" added to the language instead.

Reply:
>  during a reviews back in 2006-7 it was noted to be removed and avoided.

Maybe we should have a review again, and this time decide they do not have to be removed, and do not have to be avoided?

In 15/16 years things have changed quite a bit. Maybe we don't have to avoid say lambdas now, because they were introduced after 2007. But if they were introduced in 2006, they would have to be avoided too because of that review?

Reply:
>  just respect the request to avoid using it 

Doesn't it go two ways? Maybe you could respect the request to not avoid them?

Reply:
But that all said, didn't @dmatej remove them already as mentioned?

> I removed my static imports, and moved error messages.

Reply:
Honestly on my current experience - if you are the only one fighting against static imports, which is quite standard these days, while some people come to help, especially on the project in such painful state, it simply pushes any future help away. I don't need any response for this comment, just think about it.

Reply:
I do appreciate the help and things are evolving and changing, yet - as a person who maintains this project for more than a decade, does all backports, patches etc and contributes to this project since its inception - I think I have the right to state the fact to the first time contributor(s) - especially when it's about a minute change. There are only two language features I really don't want to see - static imports (unless there is a really good reason for it - and there rarely is) and var.